lib/uber-s3/response.rb

```diff
@@ -25,11 +25,17 @@ def success?
       raise Error::Unknown, "HTTP Response: #{status}, Body: #{body}" if status == 0
       return if status < 400 || body.to_s.empty?
 
-      # Errors are XML
-      doc = Util::XmlDocument.new(body)
+      begin
 
-      self.error_key      = doc.xpath('//Error/Code').first.text
-      self.error_message  = doc.xpath('//Error/Message').first.text
+        # Errors are XML
+        doc = Util::XmlDocument.new(body)
+
+        self.error_key      = doc.xpath('//Error/Code').first.text
+        self.error_message  = doc.xpath('//Error/Message').first.text
+
+      rescue
+        raise Error::InternalError
```

Comment:
@bmhatfield I thought `Error::Unknown` was intended for the 0 status code and `InternalError` would make sens because this seems to happen on 500s, but I can make this `Error::Unknown` also.